isic-archive-tasks/isic_archive_tasks/image.py

```diff
@@ -0,0 +1,199 @@
+import io
+import os
+import re
+import subprocess
+import tempfile
+import time
+
+from bson import ObjectId
+from celery import chord
+from celery.utils.log import get_task_logger
+import requests
+import sentry_sdk
+
+from girder.models.file import File
+
+from isic_archive_tasks import app, CredentialedGirderTask
+
+
+SUPERPIXEL_VERSION = 3.0
+logger = get_task_logger(__name__)
+
+
+@app.task(base=CredentialedGirderTask)
+def ingestImage(imageId):
+    from girder.plugins.isic_archive.models.image import Image
+    image = Image().load(imageId, force=True)
+
+    if image['ingested']:
+        logger.warning('Attempting to ingest an already ingested image.')
+        return
+
+    try:
+        imageData = Image().imageData(image)
+        image['meta']['acquisition']['pixelsY'] = imageData.shape[0]
+        image['meta']['acquisition']['pixelsX'] = imageData.shape[1]
+        image['readable'] = True
+        image['ingested'] = False
+        image['ingestionState'] = {
+            'large-image': None,
+            'superpixel-mask': None
+        }
+        Image().save(image)
+
+        callback = markImageIngested.s(image['_id'])
+        chord([generateSuperpixels.s(image['_id']),
+               generateLargeImage.s(image['_id'])])(callback)
+    except Exception:
+        sentry_sdk.capture_exception()
+        image['readable'] = False
```

Comment:
Can you set `'readable'` to `False` or `None` when the image document is first created?